### PR TITLE
feat(ingestion/looker): filter Looker dashboards by folder

### PR DESCRIPTION
**Pull request by sid-acryl**

### CODE REVIEW

**Review by jjoyce0510 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py [hunk 1]
```diff
@@ -298,6 +298,17 @@ class LookerDashboardSourceConfig(
         description="When enabled, platform instance will be added in dashboard and chart urn.",
     )
 
+    folder_path_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Filter for specific looker folder. "
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py:303]: Nitpick: Say "Allow or deny dashboards from specific folders"

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py [hunk 2]
```diff
@@ -298,6 +298,17 @@ class LookerDashboardSourceConfig(
...
+        "For example: \n"
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py:304]: This comment is super long. I'll let Harshal decide what to do here

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py:304]: seems like it's missing a space in the middle, but doesn't look too long

can probably just say "Dashboards will only be ingested if they're allowed by both this config and `dashboard_pattern`"

image

**Comment by hsheth2** [comment]
Let's add some tests for this as well :)

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py (hunk 2 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py:304]: done

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py (hunk 1 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/looker/looker_config.py]: done

**Comment by sid-acryl** [comment]
> Let's add some tests for this as well :)
> 
> can be done in a follow up

Added test-case

File: metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py [hunk 3]
```diff
@@ -1334,6 +1334,20 @@ def process_dashboard(
...
         workunits = []
+        if (
+            looker_dashboard.folder_path is not None
+            and not self.source_config.folder_path_pattern.allowed(
+                looker_dashboard.folder_path
+            )
+        ):
+            print(
+                f"Folder path {looker_dashboard.folder_path} is denied in folder_path_pattern {self.source_config.folder_path_pattern}"
+            )
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py:1345]:
```suggestion
```

**Review by hsheth2 — APPROVED** [review]

**Comment by anshbansal** [comment]
Failures are of kafka connect so ignoring them

**Pull request merged**